uint32 motor
int32 position
